yaml2requirement.py

```diff
@@ -0,0 +1,260 @@
+import re
+
+import yaml
+from colorama import Fore
+from packaging.requirements import Requirement
+
+from _helper_functions import print_color
+
+
+class YAML2Requirement():
+    _yaml_list:list = list()
+    _exclude:bool = False
+    _package_duplicates:dict = dict()
+    requirements:set = set()
+
+
+    def __init__(self, yaml_file:str, exclude:bool = False) -> None:
+        try:
+            with open(yaml_file, 'r') as f:
+                self._yaml_list = yaml.load(f, yaml.Loader)
+        except FileNotFoundError:
+            print_color(f'File not found, please check the file: {yaml_file}', Fore.RED)
+        self._exclude = exclude
+
+        self._package_duplicates = self._assemble_requirements_duplicates()
+        self.requirements = self._yaml_to_requirement(self._yaml_list, exclude=self._exclude)
+        _combine_package_duplicates(self)
+
+
+    def _change_specifier_logic(self, spec_with_text: str) -> tuple:
+        """Change specifier logic to opposite
+            e.g. "<3" will be ">3"
+            - return (new_specifier, text_after, old_specifier)
+        """
+        pattern = re.compile(r'(===|==|!=|~=|<=?|>=?|===?)\s*(.*)')
+        str_match = pattern.findall(spec_with_text)
+        specifier, text = str_match[0]
+        replacements = (('<', '>'),
+                        ('>', '<'),
+                        ('!', '='),
+                        ('~', '!'),
+                        ('==', '!='))
+        for old, new in replacements:
+            if old in specifier and specifier != '===':
+                new_spec = specifier.replace(old, new)
+                break
+            elif specifier == '===':
+                new_spec = '==='
+                break
+        return (new_spec, text, specifier)
+
+
+    def _yaml_to_requirement(self, yaml:list, exclude: bool = False) -> set:
```

Comment:
Please do a significant optimization of this function as well (based on our recent conversation).